### PR TITLE
Improvements to Controls for connecting parameters

### LINKED ISSUE #9087: Parameter connections not working as expected.
### Description of the enhancement or error report
The InputParameters::addControllableParameterConnection isn't working like expected.

### Rationale for the enhancement or information for reproducing the error
If two parameters are connect the connection isn't actually doing anything.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
It will improve the Control system.

### PR DESCRIPTION
**Pull request by aeslaughter**
This makes the name objects a bit more restrictive and adds a test that mimics the behavior you are looking for.

(closes #9087)

### CODE REVIEW

**Comment by aeslaughter** [comment]
Controls docs: http://mooseframework.org/docs/PRs/9088/site/documentation/systems/Controls/index.html

**Comment by aeslaughter** [comment]
Marmot will fail, but I have a patch waiting.

File: docs/content/documentation/systems/Controls/index.md [hunk 1]
```diff
@@ -1,7 +1,81 @@
...
+
+!!!note
+    The `declareControllable` method also accepts a space separated list of parameters.
+
+
+## Create a Control object
+
+`Control` objects are similar to other systems in MOOSE. You create a control in your application
+by inheriting from the `Control` C++ class in MOOSE. It is required to override the `execute`
+method in your custom object. within this method the following methods are generally used to get
```
Comment (andrsd) [inline docs/content/documentation/systems/Controls/index.md]: Within (capital letter).

File: docs/content/documentation/systems/Controls/index.md [hunk 2]
```diff
@@ -1,7 +1,81 @@
...
+or set controllable parameters:
+
+  * `getControllableValue` <br>
+  This method returns the current controllable parameter, in the case that multiple parameters are
+  being controlled, only the first value will be returned and a warning will be produced if the
+  values are differ (this warning may be disabled).
+
+  * `setControllableValue` <br>
+  This will method allows for a controllable parameter to be changed, in the case that multiple
```
Comment (andrsd) [inline docs/content/documentation/systems/Controls/index.md]: This method allows (drop "will").

File: docs/content/documentation/systems/Controls/index.md [hunk 3]
```diff
@@ -1,7 +1,81 @@
...
+  parameters are being controlled, all of the values will be set.
+
+There are additional methods as well is overloaded versions of these methods that allow for the
```
Comment (andrsd) [inline docs/content/documentation/systems/Controls/index.md]: "methods with(?) overloaded versions of these methods" - "is" there does not make sense to me.

Maybe you wanted to say: "There are additional overloaded methods that allow...". The sentence does not read well...

File: docs/content/documentation/systems/Controls/index.md [hunk 4]
```diff
@@ -1,7 +1,81 @@
...
+* **`object`**: specifies the input file sub-block name (e.g., "diff" in \ref{controls_example2}).
+* **`name`**: specifies the parameter name (e.g., "coef" in \ref{controls_example2}).
+
+!listing test/tests/controls/real_function_control/real_function_control.i block=Kernels id=controls_example2 caption=Example of a "Kernel" block that contains a parameter that is controlled via a [MOOSE] Control object.
+
+As shown in \ref{controls_example} an asterisk ("*") can be substituted for any one of these three
+"names", doing so allows multiple parameters to match and be controlled simultaneously.
+
+In similar fashion, object names can be defined (e.g., as in the
+g[`TimePeriod`](framework/TimePeriod.md)) object. In this case, the general name scheme is the same
```
Comment (andrsd) [inline docs/content/documentation/systems/Controls/index.md]: The leading `g` should not be there, I suppose...

File: docs/content/documentation/systems/Controls/index.md [hunk 5]
```diff
@@ -1,7 +1,81 @@
...
+as above but the parameter name is not included.
+
+In both cases there is an alternative form for defining an object and parameter names:
+`base::object/name`. In the this case "base" is the [MOOSE] base class that the object belongs.
```
Comment (andrsd) [inline docs/content/documentation/systems/Controls/index.md]: "In this case", drop the "the".
"the object belongs **to**" (?)

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by andrsd** [comment]
I see a change on the doco, so I guess I will go read the newer version...

**Comment by aeslaughter** [comment]
@andrsd There shouldn't be any changes since yesterday evening.

**Comment by andrsd** [comment]
I guess, I just had an older version of the page loaded. Well, my review comments seems to be still valid...

**Review by andrsd — APPROVED** [review]

**Pull request merged**